lib/services/elasticsearch.js

```diff
@@ -153,519 +166,479 @@ class ElasticSearch extends Service {
+   *
+   * @returns {Promise} { scrollId, hits, aggregations, total }
    */
-  search(request) {
-    const esRequest = initESRequest(request, ['from', 'size', 'scroll']);
-
-    esRequest.body = request.input.body;
-
-    if (!request.input.args.includeTrash) {
-      addActiveFilter(esRequest);
-    }
+  search (index, collection, body, { from=null, size=null, scroll=null }) {
+    const esRequest = {
+      index: this._getESIndex(index, collection),
+      body,
+      from,
+      size,
+      scroll
+    };
 
     return this.client.search(esRequest)
-      .then(result => {
-        const formatted = formatSearchResults(result);
+      .then(({ body }) => {
+        const apiResponse = {
+          scrollId: body.scrollId,
+          hits: body.hits,
+          aggregations: body.aggregations,
+          total: body.total
+        };
 
-        if (formatted.scrollId !== undefined) {
+        if (apiResponse.scrollId) {
           const
             // ms(scroll) may return undefined if in microseconds or in nanoseconds
             ttl = ms(esRequest.scroll) || ms(this.config.defaults.scrollTTL),
             key = scrollCachePrefix + this.kuzzle.constructor.hash(
-              formatted.scrollId
+              apiResponse.scrollId
             );
 
           return this.kuzzle.services.list.internalCache.psetex(key, ttl, 0)
-            .then(() => formatted);
+            .then(() => apiResponse);
         }
 
-        return formatted;
+        return apiResponse;
       })
-      .catch(error => Bluebird.reject(this.esWrapper.formatESError(error)));
+      .catch(error => this.esWrapper.reject(error));
   }
 
   /**
    * Get the document with given ID
-   * @param {Request} request
-   * @returns {Promise} resolve the document
+   *
+   * @param {string} index - Index name
+   * @param {string} collection - Collection name
+   * @param {string} id - Document ID
+   *
+   * @returns {Promise} { _id, _version, _source }
    */
-  get(request) {
-    const esRequest = initESRequest(request);
-
-    esRequest.id = request.input.resource._id;
+  get (index, collection, id) {
+    const esRequest = {
+      index: this._getESIndex(index, collection),
+      id
+    };
 
     // Just in case the user make a GET on url /mainindex/test/_search
     // Without this test we return something weird: a result.hits.hits with all document without filter because the body is empty in HTTP by default
     if (esRequest.id === '_search') {
-      return Bluebird.reject(this.getError('wrong_get_action'));
+      return this.reject('wrong_get_action');
     }
 
     return this.client.get(esRequest)
-      .then(result => {
-        if (
-          result._source
-          && result._source._kuzzle_info
-          && !result._source._kuzzle_info.active
-          && !request.input.args.includeTrash
-        ) {
-          return this.reject('document_not_found', esRequest.id);
-        }
+      .then(({ body }) => {
+        const apiResponse = {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "apiResponse". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1488)